custom/enikshay/management/commands/data_dumps_private_payments.py

```diff
@@ -0,0 +1,144 @@
+from __future__ import (
+    absolute_import,
+    print_function,
+    unicode_literals,
+)
+
+from corehq.form_processor.interfaces.dbaccessors import CaseAccessors
+
+from custom.enikshay.case_utils import (
+    CASE_TYPE_TEST,
+    CASE_TYPE_EPISODE,
+    CASE_TYPE_OCCURRENCE,
+    CASE_TYPE_PRESCRIPTION,
+    CASE_TYPE_PERSON,
+    CASE_TYPE_VOUCHER,
+    get_first_parent_of_case,
+)
+
+from custom.enikshay.const import ENROLLED_IN_PRIVATE
+from custom.enikshay.management.commands.base_data_dump import BaseDataDump
+
+DOMAIN = "enikshay"
+
+
+class IncorrectVoucherType(Exception):
+    pass
+
+
+class Command(BaseDataDump):
+    """ data dumps for private payments
+
+    https://docs.google.com/spreadsheets/d/1t6cd-VPy6p8EOEhQJD15IbULU0EJ05ALQ0tcdfx6ng8/edit#gid=1577464442&range=A43
+    """
+    TASK_NAME = "07_private_payments"
+    INPUT_FILE_NAME = "data_dumps_private_payments.csv"
+
+    def __init__(self, *args, **kwargs):
+        super(Command, self).__init__(*args, **kwargs)
+        self.case_type = CASE_TYPE_VOUCHER
+        self.case_accessor = CaseAccessors(DOMAIN)
+
+    def get_custom_value(self, column_name, case):
+        if column_name == "eNikshay person UUID":
+            person_case = self.get_person(case)
+            return person_case.case_id
+        elif column_name == "eNikshay episode UUID":
+            episode_case = self.get_episode(case)
+            return episode_case.case_id
+        raise Exception("unknown custom column %s" % column_name)
+
+    def get_case_ids_query(self, case_type):
+        """
+        ToDo: add date_fulfilled check in ES case ids query itself
```

Comment:
Do you still plan to address this `ToDo`?  I imagine you could also check the case state to see whether it's been fulfilled.

Reply:
i already added this on the case in `include_case_in_dump`.. but adding this in ES would help smaller bunch of ids..I just was not clear how to add this in ES. checking if a case property is not null or is present. Can you suggest how and if this can be added via ES query?

Reply:
Seems fine to skip.  It's doable, would just require making a new case_search filter.

I think it would basically be [`case_property_filter`](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/es/case_search.py#L106-L113), but with the `value` line using `filters.empty` instead of `filters.term`.